e2e_test/ddl/table/table.slt.part

```diff
@@ -259,3 +259,42 @@ drop table t1;
 
 statement ok
 drop table t2;
+
+statement error
+create table t(v int) with (retention_seconds = 5);
+
+statement ok
+create table t(v int) APPEND ONLY with (retention_seconds = 5);
+
+statement ok
+create index i on t(v);
+
+statement ok
+insert into t values(1);
+
+statement ok
+flush;
+
+query I
+select * from t;
+----
+1
+
+query I
+select * from i;
+----
+1
+
+statement ok
+select pg_sleep(10);
```

Comment:
QQ. Once a row has expired, but hasn't been removed by compactor yet, will the scan iterator "hides" it?

Reply:
I remember there is some code in state store implementation to "hide" it. need to check @Li0k

Reply:
We definitely hide it, by filtering it in the `min_epoch` of the iterator on the read path.